src/UI/Root/Components/Sidebar/EnvironmentControls/EnvironmentControls.tsx

```diff
@@ -1,61 +1,46 @@
 import React, { useContext, useEffect } from "react";
 import { Flex, FlexItem, Label, Stack, StackItem } from "@patternfly/react-core";
-import { RemoteData } from "@/Core";
+import { useGetEnvironmentDetails } from "@/Data/Managers/V2/Environment";
 import { DependencyContext } from "@/UI/Dependency";
 import { words } from "@/UI/words";
 import { HaltButton } from "./HaltButton";
 import { ResumeButton } from "./ResumeButton";
 
+/**
+ * This component is used to display the environment controls.
+ * It is used to halt and resume the environment.
+ * It also displays the status of the environment.
+ *
+ * @returns {React.FC}
+ */
 export const EnvironmentControls: React.FC = () => {
-  const { queryResolver, environmentHandler } = useContext(DependencyContext);
+  const { environmentHandler } = useContext(DependencyContext);
 
   const id = environmentHandler.useId();
-  const [data] = queryResolver.useContinuous<"GetEnvironmentDetails">({
-    kind: "GetEnvironmentDetails",
-    details: false,
-    id,
-  });
+  const { data, isSuccess, isError } = useGetEnvironmentDetails().useOneTime(id);
 
   useEffect(() => {
-    RemoteData.fold(
-      {
-        notAsked: () => null,
-        loading: () => null,
-        failed: () => {
-          document.dispatchEvent(new CustomEvent("status-down"));
+    if (isSuccess) {
```

Comment:
shouldn't we use the onSuccess and onError instead of the useEffect?

Reply:
There is no onSuccess/onError in useQuery, it's reserved for useMutation